### PR TITLE
Add reserve storage percentage in helm chart

### PR DESCRIPTION
**Pull request by aheyne**
#5958 

- Add the StorageReservedPercentageForDefaultDisk configuration to the helm chart.

### CODE REVIEW

**Review by innobead — DISMISSED** [review]
LGTM. 

@mantissahz @c3y1huang Please help review this as well.

**Review by mantissahz — APPROVED** [review]
LGTM

File: dev/upgrade-responder/install.sh [hunk 1]
```diff
@@ -305,6 +305,9 @@ configMap:
         "longhornSettingStorageMinimalAvailablePercentage": {
           "dataType": "float"
         },
+        "longhornSettingStorageReservedPercentageForDefaultDisk": {
```
Comment (c3y1huang) [inline dev/upgrade-responder/install.sh:308]: Already exists in https://github.com/longhorn/longhorn/pull/6258/files#diff-5a0da96ff63d171316630a8bb3a731c3287ba8db343a47c2076c0d3e75824509R314.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

**Review by c3y1huang — APPROVED** [review]
Suggest consolidating to a single commit.

**Comment by aheyne** [comment]
Rebased and squashed

**Review by mantissahz — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x

**Pull request merged**